parsers/JP.py

```diff
@@ -118,15 +121,55 @@ def fetch_consumption_df(zone_key='JP-TK', target_datetime=None,
     df = df[['datetime', 'cons']]
     return df
 
+
+def fetch_price(zone_key='JP-TK', session=None, target_datetime=None,
+                logger=logging.getLogger(__name__)):
+    if target_datetime is None:
+        target_datetime = dt.datetime.now() + dt.timedelta(days=1)
+
+    url = 'http://www.jepx.org/market/excel/spot_{}.csv'.format(
+        target_datetime.year)
```

Comment:
The year in the source url does not refer to calendar year, but to Japanese fiscal year, starting on Apr 1st. For example the the 2017 csv has prices from 2017/04/01 to 2018/03/31. Thus target_datetime needs to be shifted three months back before extracting year.

Reply:
Based on the graph of system price at http://www.jepx.org/ and the fact that it matches the last available day, column 6, of the csv source I am quite sure that the prices are half-hourly.